public/tabs/chat/chat_page_content.test.tsx

```diff
@@ -0,0 +1,232 @@
+  return { MessageContent: () => <div /> };
+});
+
+describe('<ChatPageContent />', () => {
+  const abortActionMock = jest.fn();
+  const executeActionMock = jest.fn();
+
+  beforeEach(() => {
+    jest.spyOn(chatContextExports, 'useChatContext').mockReturnValue({
+      sessionId: 'test_session_id',
+      actionExecutors: {
+        view_ppl_visualization: jest.fn(),
+      },
+      currentAccount: {
+        username: 'test_user',
+        tenant: 'private',
+      },
+    });
+
+    jest.spyOn(chatStateHookExports, 'useChatState').mockReturnValue({
+      chatState: { messages: [], llmResponding: false, interactions: [] },
+      chatStateDispatch: jest.fn(),
+    });
+
+    jest.spyOn(chatActionHookExports, 'useChatActions').mockReturnValue({
+      regenerate: jest.fn(),
+      send: jest.fn(),
+      loadChat: jest.fn(),
+      openChatUI: jest.fn(),
+      executeAction: executeActionMock,
+      abortAction: abortActionMock,
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('should display welcome message by default', () => {
+    render(<ChatPageContent messagesLoading={false} onRefresh={jest.fn()} />);
+    expect(screen.queryAllByLabelText('chat message bubble')).toHaveLength(1);
+    expect(screen.queryByLabelText('chat welcome message')).toBeInTheDocument();
+  });
+
+  it('should display a default suggested action', () => {
+    render(<ChatPageContent messagesLoading={false} onRefresh={jest.fn()} />);
+    expect(screen.queryAllByLabelText('chat suggestions')).toHaveLength(1);
+    expect(screen.queryByText('What are the indices in my cluster?')).toBeInTheDocument();
+  });
+
+  it('should display messages', () => {
+    const messages: IMessage[] = [
+      {
+        type: 'input',
+        content: 'what indices are in my cluster?',
+        contentType: 'text',
+      },
+      {
+        type: 'output',
+        content: 'here are the indices in your cluster: .alert',
+        contentType: 'markdown',
+        suggestedActions: [{ actionType: 'send_as_input', message: 'suggested action mock' }],
+      },
+    ];
+    jest.spyOn(chatStateHookExports, 'useChatState').mockReturnValue({
+      chatState: { messages, llmResponding: false, interactions: [] },
+      chatStateDispatch: jest.fn(),
+    });
+    render(<ChatPageContent messagesLoading={false} onRefresh={jest.fn()} />);
+    expect(screen.queryAllByLabelText('chat message bubble')).toHaveLength(3);
+  });
+
+  it('should only display the suggested actions of last output', () => {
+    const messages: IMessage[] = [
+      {
+        type: 'input',
+        content: 'what indices are in my cluster?',
+        contentType: 'text',
+      },
+      {
+        type: 'output',
+        content: 'here are the indices in your cluster: .kibana',
+        contentType: 'markdown',
+        suggestedActions: [{ actionType: 'send_as_input', message: 'suggested action mock' }],
+      },
```

Comment:
Shall we add more messages section? The test case is for only display suggestions for last output. We can check if the messages displayed in the previous output message.

Reply:
updated